pkg/controller/namespace/namespace_controller.go

```diff
@@ -232,26 +233,26 @@ func deleteAllContent(kubeClient client.Interface, versions *unversioned.APIVers
 	return estimate, nil
 }
 
+// maxRetriesOnConflict is set to prevent infinite loops that block the progress of a controller
+const maxRetriesOnConflict = 10
+
 // updateNamespaceFunc is a function that makes an update to a namespace
 type updateNamespaceFunc func(kubeClient client.Interface, namespace *api.Namespace) (*api.Namespace, error)
 
 // retryOnConflictError retries the specified fn if there was a conflict error
-// TODO RetryOnConflict should be a generic concept in client code
 func retryOnConflictError(kubeClient client.Interface, namespace *api.Namespace, fn updateNamespaceFunc) (result *api.Namespace, err error) {
-	latestNamespace := namespace
-	for {
-		result, err = fn(kubeClient, latestNamespace)
-		if err == nil {
-			return result, nil
-		}
-		if !errors.IsConflict(err) {
-			return nil, err
+	result = namespace
+	err = client.RetryOnConflict(wait.Backoff{Steps: maxRetriesOnConflict}, func() error {
+		if result == nil {
+			if result, err = kubeClient.Namespaces().Get(namespace.Name); err != nil {
+				return err
+			}
 		}
-		latestNamespace, err = kubeClient.Namespaces().Get(latestNamespace.Name)
-		if err != nil {
-			return nil, err
+		if result, err = fn(kubeClient, result); err != nil {
+			result = nil
 		}
-	}
+		return err
+	})
```

Comment:
Shouldn't this function be replaced by the client function now instead of just another layer of indirection?

Also, why is the order of the `fn` call and `Namespaces().Get()` reversed? And are the backoffs in the client not a good fit? You also construct a backoff per call, and elsewhere you have a package level backoff object.